chat.delta.desktop.yml

```diff
@@ -0,0 +1,120 @@
+        - "/share"
+        - "/app/lib/node_modules/npm/changelogs"
+        - "/app/lib/node_modules/npm/doc"
+        - "/app/lib/node_modules/npm/html"
+        - "/app/lib/node_modules/npm/man"
+        - "/app/lib/node_modules/npm/scripts"       
+    sources:
+        - type: archive
+          url: https://nodejs.org/dist/v8.13.0/node-v8.13.0.tar.xz
+          sha256: 2aa99474a336c6339d14f08cc27d387c5168e6fb6cbcaaea1d5ff7aa89642de2
+  
+  - name: delta
+    buildsystem: simple
+    build-options:
+        env:
+            electron_config_cache: /run/build/delta/npm-cache
+            npm_config_nodedir: /app/
+            #npm_config_devdir: /run/build/delta/npm-dev
+            npm_config_devdir: /app
+            ELECTRON_CACHE: /run/build/delta/npm-cache
+    build-commands:
+        - npm install  --offline --cache=/run/build/delta/npm-cache/  --verbose
+        - npm run build  --offline --cache=/run/build/delta/npm-cache/  --verbose
+        # This seems to be the appropriate command equivalent to a "make install".
+        # - ./node_modules/.bin/electron-builder build --linux dir
+        # It seems to be provided by the "pack" shortcut.
+        - npm run pack  --offline --cache=/run/build/delta/npm-cache/  --verbose
+        - mkdir -p /app/delta /app/bin
+        - cp -ar dist/linux-unpacked/* /app/delta/
+        # Not knowing how else to install the app into a certain prefix, we copy everything :(
+        # - cp -ra * /app/
+        - install run.sh /app/bin/
+        # FIXME: These may fall out of "pack", although I can't find them in "dist" :-/
+        - install -Dm 0644 chat.delta.desktop.desktop  /app/share/applications/chat.delta.desktop.desktop
+        - install -Dm 0644 static/chat.delta.desktop.appdata.xml  /app/share/appinfo/chat.delta.desktop.appdata.xml
+        # FIXME: Where do we find the icon?!
+        #- install -Dm 0644 /app/share/icons/HighContrast/scalable/apps/accessories-calculator.svg /app/share/icons/hicolor/scalable/apps/chat.delta.desktop.svg
+        - install -Dm 0644 chat.delta.desktop.png /app/share/icons/hicolor/256x256/apps/chat.delta.desktop.png
+        - install -Dm 0644 chat.delta.desktop.png /app/share/icons/hicolor/128x128/apps/chat.delta.desktop.png
+        - install -Dm 0644 chat.delta.desktop.png /app/share/icons/hicolor/64x64/apps/chat.delta.desktop.png
+        - install -Dm 0644 chat.delta.desktop.png /app/share/icons/hicolor/32x32/apps/chat.delta.desktop.png
+        - install -Dm 0644 chat.delta.desktop.png /app/share/icons/hicolor/16x16/apps/chat.delta.desktop.png
+        
+
+    sources:
+        - generated-sources.json
+        # Of course, not all electron packages are equal *sigh*
+        # This one downloads files at installation time.
+        # Taken from https://github.com/electron/chromedriver/issues/28#issuecomment-427839364
+        - type: file
+          url: https://github.com/electron/electron/releases/download/v3.0.0/chromedriver-v3.0.0-linux-x64.zip
```

Comment:
This one is for x86_64 only. Probably you should add electron builds for other arches, e.g. x86 and ARM.

Reply:
right.
I have no way of testing that, though.
And how does one define conditional downloads?

Maybe that was the reason I intended to put the x86 barrier via `flathub.json`...

Reply:
Just add `"only-arches": [ "x86_64" ]` and similar for other arches.
>I have no way of testing that

You can test at least i386

Reply:
I'm restricting to x86-64 only due to that binary dependency. I guess it is possible to enable other architectures, too, by downloading the chromedriver for those. But I'd rather do it later once the other issues have been tackled.

Reply:
I don't get it. Which binary dependency? `chromedriver` binaries are provided for x86_64, i386, arm and aarch64.
![image](https://user-images.githubusercontent.com/4569691/50376164-8ada7600-061a-11e9-8a38-07040a00bc4e.png)

Reply:
It should be. Try with i386 at least.

Reply:
k. these files are being downloaded now.